src/lib/engines/layerCompositor/Layer.ts

```diff
@@ -32,3 +32,39 @@ export class Layer {
     this.uvToTexture = makeMatrix3Concatenation(uvTranslation, uvScale);
   }
 }
+
+export enum LayerMaskMode {
+  None = 0,
+  Alpha = 1,
+  Luminance = 2,
+  InverseAlpha = 3,
+  InverseLuminance = 4,
+}
+
+export class LayerMask extends LayerTexture {
+  constructor(
+    compositor: LayerCompositor,
+    url: string,
+    texImage2D: TexImage2D,
+    offset: Vector2,
+    public mode: LayerMaskMode = LayerMaskMode.Alpha,
```

Comment:
I think black/white is more common than alpha for the source of the mask

Reply:
Correct. I did so.